### ISSUE TITLE
hosvd producing unwanted output

### ISSUE THREAD

**Issue by dannys4** [lanaperisa/TensorToolbox.jl#11]
When I use the `hosvd` command on a tensor (e.g. ``Y = hosvd(X);``), regardless of my suppression of output, it produces the output on the command line. Additionally, when I use the ttensor ``Y``, it produces the same output as when I create it. For example,
```
julia> X = rand(3,3,4);
julia> Y = hosvd(X);

julia> Tucker tensor of size (3, 3, 4) with core tensor of size (3, 3, 4) with orthonormal factor matrices:
ttensor.cten: 
3×3×4 Array{Float64,3}:
[:, :, 1] =
 -3.21445     0.0914684   0.0408979
  0.0876434  -0.0633035   0.0830124
  0.0219007   0.136671   -0.113072 

[:, :, 2] =
 -0.0400533  -0.197241  -0.430094
 -0.370448   -0.890895  -0.357657
  0.086002   -0.389768   0.318125

[:, :, 3] =
 -0.0443823  -0.206467  -0.012906
 -0.0424573   0.438995  -0.383421
  0.133674   -0.360386   0.120147

[:, :, 4] =
 -0.0119035   0.0517371  0.198632
  0.0731258  -0.0145422  0.126688
 -0.273783   -0.188086   0.329552

ttensor.fmat[1]:
3×3 Array{Float64,2}:
 -0.609545   0.768754   0.193579
 -0.521329  -0.204757  -0.828427
 -0.59722   -0.605881   0.525582

ttensor.fmat[2]:
3×3 Array{Float64,2}:
 -0.645296   0.713298  -0.273493
 -0.534711  -0.677422  -0.505157
 -0.545598  -0.179736   0.818546

ttensor.fmat[3]:
4×4 Array{Float64,2}:
 -0.547704   0.462936  -0.689328   -0.102653 
 -0.514351  -0.589125   0.104524   -0.614369 
 -0.467961  -0.410798  -0.0205481   0.782199 
 -0.46527    0.519487   0.716575    0.0132961Tucker tensor of size (3, 3, 4) with core tensor of size (3, 3, 4) with orthonormal factor matrices:
ttensor.cten: 
3×3×4 Array{Float64,3}:
[:, :, 1] =
 -3.21445     0.0914684   0.0408979
  0.0876434  -0.0633035   0.0830124
  0.0219007   0.136671   -0.113072 

[:, :, 2] =
 -0.0400533  -0.197241  -0.430094
 -0.370448   -0.890895  -0.357657
  0.086002   -0.389768   0.318125

[:, :, 3] =
 -0.0443823  -0.206467  -0.012906
 -0.0424573   0.438995  -0.383421
  0.133674   -0.360386   0.120147

[:, :, 4] =
 -0.0119035   0.0517371  0.198632
  0.0731258  -0.0145422  0.126688
 -0.273783   -0.188086   0.329552

ttensor.fmat[1]:
3×3 Array{Float64,2}:
 -0.609545   0.768754   0.193579
 -0.521329  -0.204757  -0.828427
 -0.59722   -0.605881   0.525582

ttensor.fmat[2]:
3×3 Array{Float64,2}:
 -0.645296   0.713298  -0.273493
 -0.534711  -0.677422  -0.505157
 -0.545598  -0.179736   0.818546

ttensor.fmat[3]:
4×4 Array{Float64,2}:
 -0.547704   0.462936  -0.689328   -0.102653 
 -0.514351  -0.589125   0.104524   -0.614369 
 -0.467961  -0.410798  -0.0205481   0.782199 
julia> Z = full(Y);

julia> Tucker tensor of size (3, 3, 4) with core tensor of size (3, 3, 4) with orthonormal factor matrices:
ttensor.cten: 
3×3×4 Array{Float64,3}:
[:, :, 1] =
 -3.21445     0.0914684   0.0408979
  0.0876434  -0.0633035   0.0830124
  0.0219007   0.136671   -0.113072 

[:, :, 2] =
 -0.0400533  -0.197241  -0.430094
 -0.370448   -0.890895  -0.357657
  0.086002   -0.389768   0.318125

[:, :, 3] =
 -0.0443823  -0.206467  -0.012906
 -0.0424573   0.438995  -0.383421
  0.133674   -0.360386   0.120147

[:, :, 4] =
 -0.0119035   0.0517371  0.198632
  0.0731258  -0.0145422  0.126688
 -0.273783   -0.188086   0.329552

ttensor.fmat[1]:
3×3 Array{Float64,2}:
 -0.609545   0.768754   0.193579
 -0.521329  -0.204757  -0.828427
 -0.59722   -0.605881   0.525582

ttensor.fmat[2]:
3×3 Array{Float64,2}:
 -0.645296   0.713298  -0.273493
 -0.534711  -0.677422  -0.505157
 -0.545598  -0.179736   0.818546

ttensor.fmat[3]:
4×4 Array{Float64,2}:
 -0.547704   0.462936  -0.689328   -0.102653 
 -0.514351  -0.589125   0.104524   -0.614369 
 -0.467961  -0.410798  -0.0205481   0.782199 
 -0.46527    0.519487   0.716575    0.0132961
```
Thanks for any additional assistance.
EDIT: Example attached.

**Comment by lanaperisa** [comment]
Hi! I don't seem to have the same problem, I tried with v1.1 and v.1.3. What Julia version are you using?

**Comment by dannys4** [comment]
Hi, I'm using Version 1.3.1 on Ubuntu Linux 18.04.4.

**Comment by lanaperisa** [comment]
Hi @dannys4, I'm sorry, I just realized I never answered your message. Were you able to solve the problem??

**Comment by dannys4** [comment]
Hi, I wasn't able to solve the problem. I suspect it's in how I have Julia set up on my system, though I'm not completely sure.

**Comment by lanaperisa** [comment]
I can see where the problem is, since displaying Tucker tensors is done by a function from this package, but I'm not able to recreate the issue. You can probably get rid of it if you go to the cloned package on your system, and in TensorToolbox/src/ttensor.jl comment out lines 837-839. The display without semicolons won't be as nice, but who cares.  Then for other types of tensors just comment out the same function in its files. 

Let me know of this is an option for you, if not, I can make a branch for you to clone.

**Comment by SteffenPL** [comment]
Hi, I came across a similar issue.
I am using JuliaPro 1.5.3 on Windows (with Juno).

When I run the following lines in the REPL, then, the display function seems to be overwritten such that
everything outputs the result from the `hosvd`.


```julia
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.    
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.5.3 (2020-11-09)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release  
|__/                   |

julia> using TensorToolbox

julia> A = rand(3,4,5)
3×4×5 Array{Float64,3}:
[:, :, 1] =
 0.655456  0.478291  0.599026  0.796995
 0.893339  0.601321  0.212705  0.301135
 0.203778  0.19092   0.670231  0.757528

...

julia> 

julia> 

julia> res = hosvd(A)
Tucker tensor of size (3, 4, 5) with core tensor of size (3, 4, 5) with orthonormal factor matricesTucker tensor of size :
(3, 4, 5) with core tensor of size (3, 4, 5) with orthonormal factor matricesttensor.cten: :

3×4×5 Array{Float64,3}:ttensor.cten: 

[:, :, 3×4×5 Array{Float64,3}:1
] =[:, :,
 1] =
 -4.0878     -4.0878      0.0631515   0.0631515  0.00728869  0.00728869  -0.0107857
-0.0107857
0.0743615    0.0743615  -0.404006   -0.404006   0.308314    0.308314     0.217618
0.217618
0.0123617    0.0123617   0.195775    0.195775   0.0114204   0.0114204    0.293634

0.293634[:, :, 2

] =[:, :,
 2] =
 -0.0243234  -0.0243234  -0.0448965  -0.0448965  -0.274312    -0.274312     0.276065
0.276065
-0.207336    -0.207336   -0.480008   -0.480008   -0.627738    -0.627738    -0.0165963
-0.0165963
-0.056878    -0.056878   -0.63423    -0.63423     0.00737679   0.00737679   0.17645

0.17645[:, :, 3

] =[:, :,
 3] =
  0.0125006   0.0125006  0.0243548  0.0243548  -0.554092  -0.554092  -0.0589377
-0.0589377
0.319587     0.319587   0.0910494  0.0910494  -0.454813  -0.454813   0.10921 
0.10921
-0.149141    -0.149141   0.56836    0.56836     0.119058   0.119058   0.244034

0.244034[:, :, 4

] =[:, :,
 4] =
  0.00563452   0.00563452  -0.520407  -0.520407   0.434229   0.434229  -0.0478419
-0.0478419
-0.182481     -0.182481    -0.137883  -0.137883  -0.305564  -0.305564   0.148526
0.148526
0.230555      0.230555     0.283063   0.283063   0.257209   0.257209   0.0366566

0.0366566[:, :, 5

] =[:, :,
 5] =
  0.0329824   0.0329824   0.11501    0.11501   -0.116291   -0.116291   -0.0139461
-0.0139461
-0.144657    -0.144657   -0.344129  -0.344129   0.0337905   0.0337905  -0.0673051
-0.0673051
0.260699     0.260699    0.243126   0.243126  -0.262474   -0.262474   -0.148511-0.148511

ttensor.fmat[1]:
3×3 Array{Float64,2}:

ttensor.fmat[1]:

 3×3 Array{Float64,2}:
-0.469101   -0.469101   0.827198   0.827198  -0.309335
-0.309335
-0.625035   -0.625035  -0.558422  -0.558422  -0.545432
-0.545432
-0.623919   -0.623919  -0.062517  -0.062517   0.778984 0.778984

ttensor.fmat[2]:
4×4 Array{Float64,2}:

ttensor.fmat[2]:

 4×4 Array{Float64,2}:
-0.45669    -0.45669    0.84884     0.84884    -0.0529697  -0.0529697  -0.260959
-0.260959
-0.525526   -0.525526  -0.0185711  -0.0185711   0.284785    0.284785    0.801483
0.801483
-0.541491   -0.541491  -0.426278   -0.426278    0.485888    0.485888   -0.537576
-0.537576
-0.471215   -0.471215  -0.312112   -0.312112   -0.824624   -0.824624   -0.0231963-0.0231963

ttensor.fmat[3]:
5×5 Array{Float64,2}:

ttensor.fmat[3]:

 5×5 Array{Float64,2}:
-0.434561   -0.434561   0.46143    0.46143   -0.649784  -0.649784   0.244246   0.244246   0.341121
0.341121
-0.498825   -0.498825   0.43646    0.43646    0.549734   0.549734   0.311538   0.311538  -0.401762
-0.401762
-0.476407   -0.476407  -0.659586  -0.659586   0.20147    0.20147    0.347621   0.347621   0.420181
0.420181
-0.440093   -0.440093  -0.370185  -0.370185  -0.414521  -0.414521  -0.286011  -0.286011  -0.644711
-0.644711
-0.37641    -0.37641    0.156505   0.156505   0.251309   0.251309  -0.800404  -0.800404   0.360582
**julia>** Tucker tensor of size (3, 4, 5) with core tensor of size (3, 4, 5) with orthonormal factor matrices:
ttensor.cten:
3×4×5 Array{Float64,3}:
[:, :, 1] =
 -4.0878      0.0631515  0.00728869  -0.0107857
  0.0743615  -0.404006   0.308314     0.217618
  0.0123617   0.195775   0.0114204    0.293634

[:, :, 2] =
 -0.0243234  -0.0448965  -0.274312     0.276065
 -0.207336   -0.480008   -0.627738    -0.0165963
 -0.056878   -0.63423     0.00737679   0.17645

[:, :, 3] =
  0.0125006  0.0243548  -0.554092  -0.0589377
  0.319587   0.0910494  -0.454813   0.10921
 -0.149141   0.56836     0.119058   0.244034

[:, :, 4] =
  0.00563452  -0.520407   0.434229  -0.0478419
 -0.182481    -0.137883  -0.305564   0.148526
  0.230555     0.283063   0.257209   0.0366566

[:, :, 5] =
  0.0329824   0.11501   -0.116291   -0.0139461
 -0.144657   -0.344129   0.0337905  -0.0673051
  0.260699    0.243126  -0.262474   -0.148511

ttensor.fmat[1]:
3×3 Array{Float64,2}:
 -0.469101   0.827198  -0.309335
 -0.625035  -0.558422  -0.545432
 -0.623919  -0.062517   0.778984

ttensor.fmat[2]:
4×4 Array{Float64,2}:
 -0.45669    0.84884    -0.0529697  -0.260959
 -0.525526  -0.0185711   0.284785    0.801483
 -0.541491  -0.426278    0.485888   -0.537576
 -0.471215  -0.312112   -0.824624   -0.0231963

ttensor.fmat[3]:
5×5 Array{Float64,2}:
 -0.434561   0.46143   -0.649784   0.244246   0.341121
 -0.498825   0.43646    0.549734   0.311538  -0.401762
 -0.476407  -0.659586   0.20147    0.347621   0.420181
 -0.440093  -0.370185  -0.414521  -0.286011  -0.644711
julia> 
```

Now, if I just want to print `A`, I get instead again the svd result:

```julia
julia> A
3×4×5 Array{Float64Tucker tensor of size ,(33},: 
4[:, :, ,1 ] =5
)  with core tensor of size (0.6554563,   40.478291,   5)0.599026 with orthonormal factor matrices:
  ttensor.cten:
0.7969953×4×5 Array
{ Float64,0.8933393}  :
0.601321[:, :, 1  ] =
0.212705   -4.0878   0.301135   0.0631515
   0.2037780.00728869    0.19092-0.0107857   
0.670231    0.07436150.757528  -0.404006

 [:, :,   2] =0.308314
      0.6574330.217618
0.221082    0.01236170.0990251     0.195775 0.834815  0.0114204
     0.6638730.293634

0.656181[:, :, 2  ] =
0.902946    -0.02432340.161321  -0.0448965
   0.671108-0.274312       0.8779380.276065
0.873434    -0.207336 0.294206  -0.480008

 [:, :,   3] =-0.627738
     0.518039-0.0165963
0.608615   -0.056878 0.379011    -0.63423  0.40928   0.00737679
    0.1667190.17645

0.92371[:, :,  3  ] =
0.701945    0.01250060.900961  0.0243548
   0.994348-0.554092    0.47734-0.0589377   
0.331458    0.319587 0.436452  0.0910494

[:, :,   4] =-0.454813
    0.8061960.10921
0.402549   -0.149141 0.247619    0.56836  0.0542718   0.119058
    0.2941520.244034

0.625626[:, :, 4  ] =
0.341086    0.005634520.872536  -0.520407
    0.331980.434229     0.508854-0.0478419
0.792934   -0.182481  0.822752  -0.137883

[:, :,   5] =-0.305564
    0.227610.148526
0.262373    0.230555  0.583312     0.2830630.0623025   0.257209
    0.1800370.0366566

0.763194[:, :, 5  ] =
0.810122    0.03298240.0642583   0.11501
    0.168881-0.116291     0.530534-0.0139461
0.891119   -0.144657 0.515975  -0.344129

julia>   -0.0673051
  0.260699    0.243126  -0.262474   -0.148511

ttensor.fmat[1]:
3×3 Array{Float64,2}:
 -0.469101   0.827198  -0.309335
 -0.625035  -0.558422  -0.545432
 -0.623919  -0.062517   0.778984

ttensor.fmat[2]:
4×4 Array{Float64,2}:
 -0.45669    0.84884    -0.0529697  -0.260959
 -0.525526  -0.0185711   0.284785    0.801483
 -0.541491  -0.426278    0.485888   -0.537576
 -0.471215  -0.312112   -0.824624   -0.0231963

ttensor.fmat[3]:
5×5 Array{Float64,2}:
 -0.434561   0.46143   -0.649784   0.244246   0.341121
 -0.498825   0.43646    0.549734   0.311538  -0.401762
 -0.476407  -0.659586   0.20147    0.347621   0.420181
 -0.440093  -0.370185  -0.414521  -0.286011  -0.644711
julia> 
```


Moreover, whenever I enter anything else into the REPL, one part of the `hosvd` output is printed again?!

```julia
julia> 1+2
3

julia> (3, 4, 5) with core tensor of size (3, 4, 5) with orthonormal factor matrices:
ttensor.cten:
3×4×5 Array{Float64,3}:
[:, :, 1] =
 -4.0878      0.0631515  0.00728869  -0.0107857
  0.0743615  -0.404006   0.308314     0.217618
  0.0123617   0.195775   0.0114204    0.293634

[:, :, 2] =
 -0.0243234  -0.0448965  -0.274312     0.276065
 -0.207336   -0.480008   -0.627738    -0.0165963
 -0.056878   -0.63423     0.00737679   0.17645

[:, :, 3] =
  0.0125006  0.0243548  -0.554092  -0.0589377
  0.319587   0.0910494  -0.454813   0.10921
 -0.149141   0.56836     0.119058   0.244034

[:, :, 4] =
  0.00563452  -0.520407   0.434229  -0.0478419
 -0.182481    -0.137883  -0.305564   0.148526
  0.230555     0.283063   0.257209   0.0366566

[:, :, 5] =
  0.0329824   0.11501   -0.116291   -0.0139461
 -0.144657   -0.344129   0.0337905  -0.0673051
  0.260699    0.243126  -0.262474   -0.148511

ttensor.fmat[1]:
3×3 Array{Float64,2}:
 -0.469101   0.827198  -0.309335
 -0.625035  -0.558422  -0.545432
 -0.623919  -0.062517   0.778984

ttensor.fmat[2]:
4×4 Array{Float64,2}:
 -0.45669    0.84884    -0.0529697  -0.260959
 -0.525526  -0.0185711   0.284785    0.801483
 -0.541491  -0.426278    0.485888   -0.537576
 -0.471215  -0.312112   -0.824624   -0.0231963

ttensor.fmat[3]:
5×5 Array{Float64,2}:
 -0.434561   0.46143   -0.649784   0.244246   0.341121
 -0.498825   0.43646    0.549734   0.311538  -0.401762
 -0.476407  -0.659586   0.20147    0.347621   0.420181
 -0.440093  -0.370185  -0.414521  -0.286011  -0.644711
julia> 
```

The problem does not occur if I run a file that contains the same commands! I tried looking into the source code, but I am not sure what exactly happens.

**Comment by lanaperisa** [comment]
Hi Steffen, that looks really ugly :/  When I run the same lines in Julia 1.5.3 in Windows the output is normal, as you would expect.

Could you maybe try the following - 

``julia> res = hosvd(A);``
``julia> display(res)``

and let me know the output? Thanks!

**Comment by SteffenPL** [comment]
Hi Lana,
thanks for the fast reply (and for the package)! 

The output is essentially the same. If you want I can try it with a normal Julia installation instead of JuliaPro. 

I also just noticed that I'm using TensorToolbox v1.0.1! (I don't like JuliaPro... arg)

**Comment by lanaperisa** [comment]
I think I know what's causing the problem, let me try some things out and get back to you.

**Comment by lanaperisa** [comment]
Hey @SteffenPL, I've released a new version with what I think could be the fix to this problem. I still wasn't able to test it, since for me, on both Windows and Linux the display is fine. Let me know if it worked.

**Comment by SteffenPL** [comment]
Hi Lana, I will test it. Somehow "] add TensorToolbox#master" "] add TensorToolbox@v1.0.3" "] update TensorToolbox"
all only install 1.0.1 for me (I don't know the specifics of the Julia registry, could it be that it's not updated there?)

Anyway, I will download it today afternoon and try it out!

**Comment by SteffenPL** [comment]
Ok, I downloaded it from git and the same problem still occurs with JuliaPro 1.5.3 :/ 

I also installed Julia 1.5.4. There it worked as expected. I suspect that it's some strange behavior related to JuliaPro. Maybe not really your packages bug, but something else? 

Maybe I can try to make a minimal example of the printing bug and submit that issue on Discourse or to JuliaPro.

(btw, in the normal Julia installation, also the older version of TensorToolbox work without the printing bug.)
